tests/main/views/test_reset_password.py

```diff
@@ -251,3 +253,191 @@ def test_should_be_an_error_if_send_email_fails(self, send_email):
 
         assert res.status_code == 503
         assert PASSWORD_RESET_EMAIL_ERROR in res.get_data(as_text=True)
+
+
+class TestChangePassword(BaseApplicationTest):
+
+    _user = None
+
+    def setup_method(self, method):
+        super().setup_method(method)
+
+        data_api_client_config = {'get_user.return_value': self.user(
+            123, "email@email.com", 1234, 'name', 'Name'
+        )}
+
+        self._user = {
+            "user": 123,
+            "email": 'email@email.com',
+        }
+
+        self.data_api_client_patch = mock.patch(
+            'app.main.views.reset_password.data_api_client', **data_api_client_config
+        )
+        self.data_api_client = self.data_api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    @pytest.mark.parametrize(
+        'user_role, redirect_url',
+        [('buyer', '/buyers'), ('supplier', '/suppliers'), ('admin', '/admin')]
+    )
+    def test_change_password_page_displays_correctly(self, user_role, redirect_url):
+        if user_role == 'buyer':
+            self.login_as_buyer()
+        elif user_role.startswith('admin'):
```

Comment:
Why not just use user_role == 'admin'?

Reply:
Because there are now five (count them!) admin roles - `admin-ccs-sourcing`, `admin-manager` and so on.

Reply:
But isn't `user-role` just coming in from the paramterization which only has `admin`?

Reply:
Haha you're right, sorry. I thought this was from the helper function we added (https://github.com/alphagov/digitalmarketplace-user-frontend/blob/master/app/main/helpers/login_helpers.py#L24). Yep this can be changed 😸

Reply:
Perhaps if there are two things here then we need `user_role` and `user_role_type` or _something_

Reply:
Nah I think it was just a poor choice of parameter - it should be `admin-ccs-sourcing` or similar, as we no longer use a straight `admin` role. I'll update it to be 💯clear.

Reply:
My mistake - we do still have `admin` as a valid role. Going to keep as is for now.